R/create.R

```diff
@@ -500,37 +498,23 @@ create_stan_data <- function(reported_cases, generation_time,
 create_initial_conditions <- function(data) {
   init_fun <- function() {
     out <- list()
-    if (data$n_uncertain_mean_delays > 0) {
-      out$delay_mean <- array(purrr::map2_dbl(
-        data$delay_mean_mean[data$uncertain_mean_delays],
-        data$delay_mean_sd[data$uncertain_mean_delays] * 0.1,
-        ~ rnorm(1, mean = .x, sd = .y)
-      ))
-    }
-    if (data$n_uncertain_sd_delays > 0) {
-      out$delay_sd <- array(purrr::map2_dbl(
-        data$delay_sd_mean[data$uncertain_sd_delays],
-        data$delay_sd_sd[data$uncertain_sd_delays] * 0.1,
-        ~ rnorm(1, mean = .x, sd = .y)
-      ))
-    }
-    if (data$truncation > 0) {
-      if (data$trunc_mean_sd > 0) {
-        out$truncation_mean <- array(rnorm(1,
-          mean = data$trunc_mean_mean,
-          sd = data$trunc_mean_sd * 0.1
-        ))
-      }
-      if (data$trunc_sd_sd > 0) {
-        out$truncation_sd <- array(
-          truncnorm::rtruncnorm(1,
-            a = 0,
-            mean = data$trunc_sd_mean,
-            sd = data$trunc_sd_sd * 0.1
-          )
-        )
-      }
-    }
+    out$delay_mean <- array(purrr::map2_dbl(
```

Comment:
Multiple repeated chunks here now that could/should really be coming from a small function.

Reply:
`truncnorm` is vectorized, and this is generating a scalar anyway, so the `array(purrr::map2_dbl(...))` seems overkill.

imo, the nice syntax here would be `with(X, truncnorm(1, a = 0, mean = mean, sd = sd))`. that would mean structuring data differently. more like `data$delay$mean$mean, data$delay$mean$sd, ...`.

that sort of approach would also be in the direction of something like `with(data$delay$mean, do.call(rdistro, c(args, list(n=1)))` which is the sort of maximum flexibility option.

Reply:
given how systematically names are structured currently, this should be an `apply`-family-able operation, if structured correctly. i.e. "for all of these set of names (delay, trunc, gt): for their sub-elements (mean, sd), use the sub-sub elements (mean, sd) this way"

Reply:
This sounds like someone would love to write a little helper function to improve our DRY  🤡

Reply:
...trying to get me fired for distraction by shiny objects.

Reply:
ah it will just be a formal reprimand - got to be worth it for all the fame and adulation this will bring.

Reply:
removed `map2_dbl` call in https://github.com/epiforecasts/EpiNow2/pull/363/commits/145c3f8a20492fd35b5b1059f0d84d81be083a9b